prescient/stats/hourly_stats.py

```diff
@@ -138,121 +142,93 @@ def __init__(self, options, day: date, hour: int):
-                    self.renewable_gen_cleared_DA[g]*price_DA + \
-                    (self.observed_renewables_levels[g] - self.renewable_gen_cleared_DA[g])*price_RT
-
-
-
-        r_price_DA = self.planning_reserve_price
-        r_price_RT = self.reserve_RT_price
-        self.thermal_reserve_revenue = { g : self.thermal_reserve_cleared_DA[g]*r_price_DA + \
-                                                ( self.observed_thermal_headroom_levels[g] - \
-                                                   self.thermal_reserve_cleared_DA[g] )*r_price_RT
-                                         for g in extractor.get_thermal_generators(sced) }
-
-        ## TODO: calculate uplift for the day
-        self.thermal_uplift = { g : 0. for g in extractor.get_thermal_generators(sced) }
-
-        ## TODO: calculate uplift for the day
-        self.renewable_uplift = { g : 0. for g in extractor.get_nondispatchable_generators(sced) }
-
+        self.thermal_fleet_capacity = ops_stats.thermal_fleet_capacity
+
+        # Add scalar fields
+        summing_fields = [
+            'sced_runtime',
+            'total_demand',
+            'fixed_costs',
+            'variable_costs',
+            'power_generated',
+            'load_shedding',
+            'over_generation',
+            'reserve_shortfall',
+            'available_reserve',
+            'available_quickstart',
+            'renewables_available',
+            'renewables_used',
+            'renewables_curtailment',
+            'on_offs',
+            'sum_on_off_ramps',
+            'sum_nominal_ramps',
+            'quick_start_additional_costs',
+            'quick_start_additional_power_generated',
+            'reserve_requirement',
+            'price', # ?
+            'reserve_RT_price', # ?
+            'planning_reserve_price' # ?
+           ]
+        for field in summing_fields:
+            val = getattr(ops_stats, field)
+            add_to = getattr(self, field)
+            setattr(self, field, add_to+val)
+
+        # Add values indexed by a model entity
+        keyed_summing_fields = [
+            'observed_thermal_dispatch_levels',
+            'observed_thermal_headroom_levels',
+            'observed_thermal_states',
+            'observed_costs',
+            'observed_renewables_levels',
+            'observed_renewables_curtailment',
+            'observed_flow_levels',
+            'bus_demands',
+            'observed_bus_mismatches',
+            'observed_bus_LMPs',
+            'storage_input_dispatch_levels',
+            'storage_output_dispatch_levels',
+            'storage_soc_dispatch_levels',
+            'planning_energy_prices',
+            'thermal_gen_cleared_DA',
+            'thermal_gen_revenue',
+            'thermal_reserve_cleared_DA',
+            'thermal_reserve_revenue',
+            'thermal_uplift',
+            'renewable_gen_cleared_DA',
+            'renewable_gen_revenue',
+            'renewable_uplift',
+           ]
```

Comment:
Similar to above...I'll work on this.

Reply:
Averaged or summed after multiplying by `sced_time_period_length_minutes/60.` (should be the same operation):
- observed_thermal_dispatch_levels
- observed_thermal_headroom_levels
- observed_thermal_states
- observed_renewables_levels
- observed_renewables_curtailment
- observed_flow_levels
- bus_demands
- observed_bus_mismatches
- observed_bus_LMPs
- storage_input_dispatch_levels
- storage_output_dispatch_levels
- storage_soc_dispatch_levels
- planning_energy_prices
- thermal_gen_cleared_DA
- thermal_reserve_cleared_DA
- renewable_gen_cleared_DA

Summed as is:
- observed_costs
- thermal_gen_revenue
- thermal_reserve_revenue
- thermal_uplift
- renewable_gen_revenue
- renewable_uplift

As an aside, since we're looking at this: the uplift calculations currently do nothing because that is rolled-up daily based on the observed revenue for the day (#20).